var/spack/repos/builtin/packages/kokkos/package.py

```diff
@@ -2,231 +2,245 @@
+        'qthread': [False, 'Eenable the QTHREAD library'],
+        'tests': [False, 'Build for tests'],
+    }
+
+    amd_gpu_arches = [
+        'fiji',
+        'gfx901',
+        'vega900',
+        'vega906',
+    ]
+    variant("amd_gpu_arch", default='none', values=amd_gpu_arches,
+            description="AMD GPU architecture")
+    conflicts("+hip", when="amd_gpu_arch=none")
+
+    spack_micro_arch_map = {
+        "aarch64": "",
+        "arm": "",
+        "ppc": "",
+        "ppc64": "",
+        "ppc64le": "",
+        "ppcle": "",
+        "sparc": None,
+        "sparc64": None,
+        "x86": "",
+        "x86_64": "",
+        "thunderx2": "THUNDERX2",
+        "k10": None,
+        "zen": "ZEN",
+        "bulldozer": "",
+        "piledriver": "",
+        "zen2": "ZEN2",
+        "steamroller": "KAVERI",
+        "excavator": "CARIZO",
+        "a64fx": "",
+        "power7": "POWER7",
+        "power8": "POWER8",
+        "power9": "POWER9",
+        "power8le": "POWER8",
+        "power9le": "POWER9",
+        "i686": None,
+        "pentium2": None,
+        "pentium3": None,
+        "pentium4": None,
+        "prescott": None,
+        "nocona": None,
+        "nehalem": None,
+        "sandybridge": "SNB",
+        "haswell": "HSW",
+        "mic_knl": "KNL",
+        "cannonlake": "SKX",
+        "cascadelake": "SKX",
+        "westmere": "WSM",
+        "core2": None,
+        "ivybridge": "SNB",
+        "broadwell": "BDW",
+        "skylake": "SKX",
+        "icelake": "SKX",
+        "skylake_avx512": "SKX",
+    }
+
+    spack_cuda_arch_map = {
+        "30": 'kepler30',
+        "32": 'kepler32',
+        "35": 'kepler35',
+        "37": 'kepler37',
+        "50": 'maxwell50',
+        "52": 'maxwell52',
+        "53": 'maxwell53',
+        "60": 'pascal60',
+        "61": 'pascal61',
+        "70": 'volta70',
+        "72": 'volta72',
+        "75": 'turing75',
+    }
+    cuda_arches = spack_cuda_arch_map.values()
+    conflicts("+cuda", when="cuda_arch=none")
+
+    devices_values = list(devices_variants.keys())
+    for dev in devices_variants:
+        dflt, desc = devices_variants[dev]
+        variant(dev, default=dflt, description=desc)
+
+    options_values = list(options_variants.keys())
+    for opt in options_values:
+        if "cuda" in opt:
+            conflicts('+%s' % opt, when="~cuda",
+                      msg="Must enable CUDA to use %s" % opt)
+        dflt, desc = options_variants[opt]
+        variant(opt, default=dflt, description=desc)
+
+    tpls_values = list(tpls_variants.keys())
+    for tpl in tpls_values:
+        dflt, desc = tpls_variants[tpl]
+        variant(tpl, default=dflt, description=desc)
+        depends_on(tpl, when="+%s" % tpl)
+
+    variant("wrapper", default=False,
+            description="Use nvcc-wrapper for CUDA build")
+    depends_on("kokkos-nvcc-wrapper", when="+wrapper")
+    conflicts("+wrapper", when="~cuda")
```

Comment:
Speaking of other things I noticed too late (i.e., after merge). What's the point of `wrapper` variant? Shouldn't  this simply be
```
depends_on("kokkos-nvcc-wrapper", when="+cuda")
```

I encountered this when working to update arborx:
```
$ spack install arborx@0.9-beta+cuda ^kokkos cuda_arch=61
 -   ti623cj  arborx@0.9-beta%gcc@7.4.0 build_type=RelWithDebInfo +cuda~hip+mpi~openmp+serial arch=linux-ubuntu18.04-broadwell   
 -   h5te24i      ^kokkos@3.1%gcc@7.4.0~aggressive_vectorization amd_gpu_arch=none build_type=RelWithDebInfo ~compiler_warnings+cuda cuda_arch=70 +cuda_lambda~cuda_ldg_intrinsic~cuda_relocatable_device_code~cuda_uvm~debug~debug_bounds_check~debug_dualview_modify_check~deprecated_code~examples~explicit_instantiation~hip~hpx~hpx_async_dispatch~hwloc~memkind~numactl~openmp+profil
ing~profiling_load_print~pthread~qthread+serial+shared std=11 ~tests~wrapper arch=linux-ubuntu18.04-broadwell 
<snip>
  >> 13    CMake Error at cmake/kokkos_test_cxx_std.cmake:109 (MESSAGE):                                                         
     14      Invalid compiler for CUDA.  The compiler must be nvcc_wrapper or Clang, but     
```
So, having `kokkos+cuda` does not enable `+wrapper`, and in general, `+wrapper` seems unnecessary. I could be missing something.

Reply:
Oh, wait, is that because you could compile CUDA with Clang? I retract the point.

Reply:
Yep, you got it : )